keps/sig-node/4639-oci-volume-source/README.md

```diff
@@ -0,0 +1,967 @@
+This allows users to package files and share them among containers in a pod without including them in the main image,
+thereby reducing vulnerabilities and simplifying image creation.
+
+While OCI images are well-supported by Kubernetes and CRI, extending support to
+OCI artifacts involves recognizing additional media types within container
+runtimes, implementing custom lifecycle management, resolution of artifact
+registry referrers use pattern for artifacts, and ensuring appropriate
+validation and security measures.
+
+## Motivation
+
+Supporting OCI images and artifacts directly as a `VolumeSource` allows
+Kubernetes to focus on OCI standards as well as allows to store and distribute
+any content using OCI registries. This allows the project to grow into use cases
+which go beyond running particular images.
+
+### Goals
+
+- Introduce a new `VolumeSource` type that allows mounting OCI images and/or artifacts.
+- Simplify the process of sharing files among containers in a pod.
+
+### Non-Goals
+
+- This proposal does not aim to replace existing `VolumeSource` types.
+- This proposal does not address other use cases for OCI objects beyond file sharing among containers in a pod.
+
+## Proposal
+
+We propose to add a new `VolumeSource` that supports OCI images and/or artifacts. This `VolumeSource` will allow users to mount an OCI object
+directly into a pod, making the files within the image accessible to the containers without the need to include them in the main image and to be able to host them in OCI compatible registries.
+
+### User Stories (Optional)
+
+#### Story 1
+
+As a Kubernetes user, I want to share a configuration file among multiple containers in a pod without including the file in my main image, so that I can
+minimize security risks and image size. 
+
+Beside that, I want:
+- to package this file in an OCI object to take advantage of OCI distribution.
+- the image to be downloaded with the same credentials that kubelet using for other images.
+- to be able to use image pull secrets when downloading the image if an image is from the registry that requires image pull secrets.
+
+#### Story 2
+
+As a DevOps engineer, I want to package and distribute binary artifacts using OCI image and distribution specification standards and mount them directly into my Kubernetes pods,
+so that I can streamline my CI/CD pipeline. I want to package this file in an OCI object to take advantage of OCI distribution.
+
+#### Story 3
+
+As a data scientist, MLOps engineer, or AI developer, I want to mount large language models or machine learning models in a pod alongside a model-server, so that I can efficiently serve the models
+without including them in the container image. I want to package these models in an OCI object to take advantage of OCI distribution and ensure
+efficient model deployment. This allows to separate the data from the executables that process them.
```

Comment:
"data" might be controversial since some may think that models are code. For this specific use case, perhaps we can clarify this to be "model specifications/content"?